Source/Details/ASDataController.mm

```diff
@@ -653,7 +657,15 @@ - (void)updateWithChangeSet:(_ASHierarchyChangeSet *)changeSet
     // Step 4: Inform the delegate on main thread
     [_mainSerialQueue performBlockOnMainThread:^{
       as_activity_scope_leave(&preparationScope);
+      if (_deallocDelegate.isDeallocating) {
+        as_log_debug(ASCollectionLog(), "datacontroller batchupdate during dealloc");
+        return;
+      }
       [_delegate dataController:self updateWithChangeSet:changeSet updates:^{
```

Comment:
Another question is, should we move the `isDeallocating` check to the implementation of the delegate? That means that `ASDataController` wouldn't mind about whether or not its delegate is being de-allocated, as long as the delegate is still around.